synapseutils/sync.py

```diff
@@ -1021,9 +1130,50 @@ def syncToSynapse(
         _manifest_upload(syn, df)
 
 
-def _manifest_upload(syn, df):
+def _convert_cell_in_manifest_to_python_types(
+    cell: str,
+) -> typing.Union[typing.List, datetime.datetime, float, int, bool, str]:
+    """
+    Takes a possibly semi-colon delimited cell from the manifest TSV file into a list
+    of items to be used as annotations.
+
+    Args:
+        cell: The cell item to convert.
+
+    Returns:
+        The list of items to be used as annotations. Or a single instance if that is
+            all that is present.
+    """
+    values_to_return = []
+
+    cell_values = cell.split(";")
+    for annotation_value in cell_values:
+        if possible_datetime := datetime_or_none(annotation_value):
+            values_to_return.append(possible_datetime)
+        elif (possible_int := int_or_none(annotation_value)) is not None:
+            values_to_return.append(possible_int)
+        elif (possible_float := float_or_none(annotation_value)) is not None:
```

Comment:
You may want to flip these.  Technically you can do something like

```
int(3.4)
```
without an error

Reply:
Great point. I will review this logic.

Reply:
I swapped to using https://docs.python.org/3/library/ast.html#ast.literal_eval here instead of doing my own parsing